src/magic_folder/status.py

```diff
@@ -311,6 +348,57 @@ def _marshal_event_poll_completed(folder_name, poller):
     return _marshal_event("poll-completed", msg)
 
 
+
+def _invite_to_json(invite):
+    return {
+        "uuid": invite.uuid,
+        "participant-name": invite.participant_name,
+        "mode": invite.participant_mode,
+    }
+
+
+def _marshal_event_invite_created(folder_name, invite):
+    msg = _invite_to_json(invite)
+    msg["folder"] = folder_name
+    return _marshal_event("invite-created", msg)
+
+
+def _marshal_event_invite_updated(folder_name, invite, **kwargs):
+    for k in kwargs.keys():
+        if k not in {"welcome", "code", "versions"}:
+            raise ValueError("Unknown extra {}".format(k))
+    msg = _invite_to_json(invite)
+    msg["folder"] = folder_name
+    msg.update(kwargs)
+    return _marshal_event("invite-updated", msg)
+
+
+def _marshal_event_invite_succeeded(folder_name, invite):
+    msg = _invite_to_json(invite)
+    msg["folder"] = folder_name
+    return _marshal_event("invite-succeeded", msg)
+
+
+def _marshal_event_invite_failed(folder_name, invite, reason):
+    msg = _invite_to_json(invite)
+    msg["folder"] = folder_name
+    msg["reason"] = reason
+    return _marshal_event("invite-failed", msg)
+
+
+def _marshal_event_invite_rejected(folder_name, invite, reason):
+    msg = _invite_to_json(invite)
+    msg["folder"] = folder_name
+    msg["reason"] = str(reason)
+    return _marshal_event("invite-rejected", msg)
+
+
+def _marshal_event_invite_cancelled(folder_name, invite):
+    msg = _invite_to_json(invite)
+    msg["folder"] = folder_name
+    return _marshal_event("invite-canceled", msg)
```

Comment:
Everywhere else this is `invite-cancelled` (with two lowercase "L"s) so this should be `invite-cancelled` too (or, alternatively, everything else should be updated to match this one).